src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/events/PluginRerunCheckEvent.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2021 Intel . All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.sonyericsson.hudson.plugins.gerrit.trigger.hudsontrigger.events;
+
+import java.io.Serializable;
+
+import org.jenkinsci.Symbol;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import com.sonyericsson.hudson.plugins.gerrit.trigger.Messages;
+import com.sonymobile.tools.gerrit.gerritevents.dto.events.RerunCheck;
+
+import hudson.Extension;
+import hudson.model.Descriptor;
+import jenkins.model.Jenkins;
+
+/**
+ * An event configuration that causes the build to be triggered when a change's
+ * WIP state changed.
+ */
+public class PluginRerunCheckEvent extends PluginGerritEvent implements Serializable {
```

Comment:
We should point out that this event is not generated by Gerrit but from the checks plugin.
That means people should install the checks plugin on Gerrit to have it working.

Reply:
There is code available to check that a plugin is installed on the Gerrit side that should be checked as well.

Reply:
But should we really care whether or not the checks-plugin is installed (and spent time and efforts to check)?

I mean this is checks-plugin specific event: [1] and is triggered iff the plugin is installed and a user klicked Re-Run on change screen button.

[1] https://gerrit-review.googlesource.com/c/plugins/checks/+/255373

Reply:
But a `help.html` next to the `config.jelly` explaining what it is and what you need would be good. I think that would satisfy @lucamilanesio 's concern.